### PR TITLE
Hotlistid 741888

### PR DESCRIPTION
**Pull request by sramamoorthy73**
1) Fix race (potential).
2) Add tests to test throttling, while read/write.

### CODE REVIEW

File: data/data.go [hunk 1]
```diff
@@ -83,6 +95,7 @@ func (d *ApidDb) Begin() (apid.Tx, error) {
 	d.mutex.Lock()
 	tx, err := d.db.Begin()
 	if err != nil {
+		d.mutex.Unlock()
```
Comment (haomingzhang) [inline data/data.go:98]: That's a good catch!

File: data/data_test.go [hunk 2]
```diff
@@ -123,6 +127,8 @@ var _ = Describe("Data Service", func() {
 		for i := 0; i < count; i++ {
 			<-finished
```
Comment (haomingzhang) [inline data/data_test.go:129]: Could you also add `Expect` here?

File: data/data_test.go [hunk 3]
```diff
@@ -105,10 +106,13 @@ var _ = Describe("Data Service", func() {
 		for i := 0; i < count+1; i++ {
 			<-finished
```
Comment (haomingzhang) [inline data/data_test.go:107]: Could you also add `Expect` here?

File: data/data.go (hunk 1 above)
Comment (sramamoorthy73) [reply data/data.go:98]: thx

File: data/data_test.go (hunk 3 above)
Comment (sramamoorthy73) [reply data/data_test.go:107]: done.

File: data/data_test.go (hunk 2 above)
Comment (sramamoorthy73) [reply data/data_test.go:129]: done.

**Review by sramamoorthy73 — COMMENTED** [review]
Done with review comment update.

**Pull request merged**